config/file.go

```diff
@@ -0,0 +1,122 @@
+package config
```

Comment:
does the filename need to be e.g., `file_provider`?

Reply:
Not strictly. All .go files in a package are pulled into the build process and their filenames have no special meaning for `go build` or `go install` or `go get`. Files ending in `_test.go` have a special meaning for `go test`, but that's the only instance I can think of at the moment. 😄

Reply:
If you would like it to be file_provider.go, I'm happy to rename it!